app/Main.hs

```diff
@@ -22,31 +22,30 @@ import Svg.Parser (parsePrebuiltSvgs)
 import Css.Compiler (compileCSS)
 import Util.Documentation (generateDocs)
 
--- dynamicgraph_test
-import DynamicGraphs.WriteRunDot(doDots)
-import DynamicGraphs.GraphGenerator(sampleGraph)
+-- dynamic graph generation
+import DynamicGraphs.WriteRunDot(generatePrereqsForCourses)
 
 -- | A map of command-line arguments to their corresponding IO actions.
-taskMap :: Map.Map String (IO ())
+taskMap :: Map.Map String ([String] -> IO ())
 taskMap = Map.fromList [
-    ("server", runServer),
-    ("database", setupDatabase),
-    ("graphs", parsePrebuiltSvgs),
-    ("css", compileCSS),
-    ("docs", generateDocs),
-    ("minfangraph", doDots [ ("sample", sampleGraph) ])]
+    ("server", const runServer),
+    ("database", const setupDatabase),
+    ("graphs", const parsePrebuiltSvgs),
+    ("css", const compileCSS),
+    ("docs", const generateDocs),
+    ("generate", \(name : courses) -> generatePrereqsForCourses (name, courses))]
 
 -- | Courseography entry point.
 main :: IO ()
 main = do
     args <- getArgs
     let taskName = if null args then "server" else head args
-    fromMaybe putUsage (Map.lookup taskName taskMap)
+    fromMaybe putUsage (Map.lookup taskName taskMap) $ tail args
```

Comment:
I think the `tail` here fails if `args` is empty? I'm a little surprised you didn't run into this, so maybe I'm mistaken. But if I'm right you could pattern-match on `case args of ...`